landlab/components/overland_flow/generate_overland_flow_deAlmeida.py

```diff
@@ -380,11 +381,14 @@ def rainfall_intensity(self):
         return self._rainfall_intensity
 
     @rainfall_intensity.setter
-    def rainfall_intensity(self, rainfall_intensity):
-        if rainfall_intensity >= 0:
-            self._rainfall_intensity = rainfall_intensity
+    def rainfall_intensity(self, new_val):
+        message = "Rainfall intensity must be positive"
+        if isinstance(new_val, float):
+            assert new_val >= 0, message
         else:
-            raise ValueError("Rainfall intensity must be positive")
+            assert (new_val >= 0).all(), message
+
+        self._rainfall_intensity = return_array_at_node(self._grid, new_val)
```

Comment:
I'm not sure you need to use ``return_array_at_node`` here. Can you not just have,
```python
self._rainfall_intensity = new_val
```